Sources/SocksCore/Error.swift

```diff
@@ -70,3 +70,144 @@ public struct Error: ErrorProtocol, CustomStringConvertible {
         return "Socket failed with code \(self.number) [\(self.type)] \"\(getReason())\""
     }
 }
+
+public struct ErrorLookUpTabel{
+    
+    public static func getCorrespondingErrorString(errorCode: Int) -> String {
+        // Dictionary indexed from zero
```

Comment:
You can actually simplify this, as it's a Dictionary, which just returns `nil` for unknown keys, use that to your advantage, and don't check the index at all. Just do

```swift
guard let description = errorDescriptions[errorCode] else { return "?" }
return description
```

This way you also get rid of the force unwrap (`!`) at the end